pkg/resolver/client/ens/ens.go

```diff
@@ -121,10 +139,44 @@ func (c *Client) Close() error {
 	return nil
 }
 
-func wrapResolve(backend bind.ContractBackend, name string) (string, error) {
+func wrapDial(endpoint string, contractAddr string) (*ethclient.Client, *goens.Registry, error) {
+
+	// Dial the eth client.
+	ethCl, err := ethclient.Dial(endpoint)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Obtain the ENS registry.
+	registry, err := goens.NewRegistryAt(ethCl, common.HexToAddress(contractAddr))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Ensure that the ENS registry client is deployed to the given contract address.
+	_, err = registry.Owner("")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return ethCl, registry, nil
+}
+
+func wrapResolve(registry *goens.Registry, addr common.Address, name string) (string, error) {
+
+	// Ensure the name is registered.
+	ownerAddress, err := registry.Owner(name)
+	if err != nil {
+		return "", err
+	}
+
+	// If the name is not registered, return an error.
+	if bytes.Equal(ownerAddress.Bytes(), goens.UnknownAddress.Bytes()) {
+		return "", errors.New("name is not registered")
```

Comment:
extract the error to a package level variable, then
```suggestion
		return "", fmt.Errorf("wrapresolve: %w", errNameNotRegistered)
```